src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -66,4 +71,66 @@ public void updateCAPassphrase(String passphrase) {
         caPassphrase.withValue(passphrase);
     }
 
-}
\ No newline at end of file
+    /**
+     * Retrieves opaque client device thing configuration.
+     *
+     * @param thingName     Thing name
+     * @param schemaVersion Thing config schema version
+     * @return config topic mapped to given thing name
+     */
+    public Topics getClientDeviceThing(String thingName, String schemaVersion) {
+        return config.findTopics(CLIENT_DEVICE_THINGS_KEY, schemaVersion, thingName);
+    }
+
+    /**
+     * Create or replace client device thing details in the config.
+     *
+     * @param thingName     client device thing name
+     * @param thingDetails  thing details
+     * @param schemaVersion Thing config schema version
+     */
+    public void createOrReplaceClientDeviceThing(String thingName,
+                                                 Map<String, Object> thingDetails,
+                                                 String schemaVersion) {
+        Topics clientDeviceThing = config.lookupTopics(CLIENT_DEVICE_THINGS_KEY, schemaVersion, thingName);
+        clientDeviceThing.replaceAndWait(thingDetails);
+    }
+
+    /**
+     * Update client device thing details in the config.
+     *
+     * @param thingName     client device Thing name
+     * @param thingDetails  map of client device Thing details to be updated
+     * @param schemaVersion Thing config schema version
+     */
+    public void updateClientDeviceThing(String thingName, Map<String, Object> thingDetails, String schemaVersion) {
+        Topics clientDeviceThing = config.findTopics(CLIENT_DEVICE_THINGS_KEY, schemaVersion, thingName);
+        if (clientDeviceThing == null) {
+            return;
+        }
+        thingDetails.forEach((key, value) -> updateClientDeviceThingDetail(clientDeviceThing, key, value));
+    }
+
+    @SuppressWarnings("PMD.AvoidBranchingStatementAsLastInLoop")
+    private void updateClientDeviceThingDetail(Topics clientDeviceThing, String detailKey, Object newValue) {
+        Topic thingDetail = clientDeviceThing.lookup(detailKey);
+        IllegalArgumentException ex = new IllegalArgumentException(
+                String.format("Unsupported thing detail value: %s", newValue));
+        if (newValue instanceof Integer) {
+            thingDetail.withValue((Integer) newValue);
+        } else if (newValue instanceof String) {
+            thingDetail.withValue((String) newValue);
+        } else if (newValue instanceof List<?>) {
+            for (Object obj : (List<?>) newValue) {
+                if (obj instanceof String) {
+                    thingDetail.withValue((List<String>) newValue);
+                } else {
+                    throw ex;
+                }
+                break;
+            }
+        } else {
+            throw ex;
+        }
+    }
```

Comment:
- This update api will let us update only the changed thing detail. Currently its not used, I'll create a separate PR for the adapter to figure out what thing details changed and call this api to only update that changed detail.
- Considering the `Topic` lets us update [only (int, String, boolean, List\<String\>) values](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/config/Topic.java#L139-L157), the `updateClientDeviceDetail()` logic may seem _somewhat_ ugly! I'm open to suggestions.